scheduler/scheduler.go

```diff
@@ -42,29 +45,24 @@ type ScheduledResource struct {
 	RequiredBy []*ScheduledResource
 	Status     ScheduledResourceStatus
 	resources.Resource
+	// parentKey -> dependencyMetadata
+	Meta map[string]map[string]string
 	sync.RWMutex
 }
 
-// IsBlocked checks whether d can be created. It checks cached resource status,
-// unless the requirement is a Service - it gets the status from the API in that case
+// IsBlocked checks whether d can be created. It checks status of resources,
+// it depends on, via API
 func (d *ScheduledResource) IsBlocked() bool {
 	isBlocked := false
 
 	for _, r := range d.Requires {
 		if isBlocked {
 			break
 		}
-		r.RLock()
-		switch r.Resource.(type) {
-		case resources.Service:
-			status, err := r.Resource.Status()
-			if err != nil || status != "ready" {
-				isBlocked = true
-			}
-		default:
-			isBlocked = r.Status != Ready
+		status, err := r.Resource.Status(r.Meta[d.Key()])
```

Comment:
well, i intended to do drop cache, since for each dependency status may be different, according to metadata, but for the case of no meta at all we could save some time. i'll put it back